internal/k8s/controller.go

```diff
@@ -2083,6 +2073,10 @@ func newVirtualServerRouteErrorFromVSR(virtualServerRoute *conf_v1.VirtualServer
 }
 
 func (lbc *LoadBalancerController) createVirtualServer(virtualServer *conf_v1.VirtualServer) (*configs.VirtualServerEx, []virtualServerRouteError) {
+	if !lbc.HasCorrectIngressClass(virtualServer) {
```

Comment:
I don't think we are on the same page with the problem with `createVirtualServer`.

the problem is that in the function we do `obj, exists, err := lbc.virtualServerRouteLister.GetByKey(vsrKey)` and we don't check the class of that VSR!

So If you create the following VSR with an incorrect class:
```yaml
apiVersion: k8s.nginx.org/v1
kind: VirtualServerRoute
metadata:
  name: coffee
spec:
  ingressClassName: random
  host: cafe.example.com
  upstreams:
  - name: latte
    service: latte-svc
    port: 80
  - name: espresso
    service: espresso-svc
    port: 80
  subroutes:
  - path: "/coffee/latte"
    action:
      pass: latte
  - path: "/coffee/espresso"
    action:
      pass: espresso
```
And then create a VS with the correct class:
```yaml
apiVersion: k8s.nginx.org/v1
kind: VirtualServer
metadata:
  name: cafe
spec:
  ingressClassName: nginx
  host: cafe.example.com
  upstreams:
  - name: tea
    service: tea-svc
    port: 80
  - name: coffee
    service: coffee-svc
    port: 80
  routes:
  - path: "/tea"
    action:
      pass: tea
  - path: "/coffee"
    route: coffee
```

The Ingress Controller will handle the VSR :
```
Events:
  Type    Reason          Age   From                      Message
  ----    ------          ----  ----                      -------
  Normal  AddedOrUpdated  6m9s  nginx-ingress-controller  Configuration for default/coffee was added or updated
```

This is a bug :(

The fix in this line doesn't fix it and isn't necessary.